code/common/subscriptions.q

```diff
@@ -65,27 +65,45 @@ replay0:{[tabs;realsubs;schemalist;logfilelist;filters]
   if[count where nullschema:0=count each schemalist;
     tabs:(schemalist where not nullschema)[;0];
     subtabs:tabs inter realsubs[`subtabs]];
+  // function to replay tplogs and apply relevant filters if required.
+  filterfunc:{[lf;td;logmetatab]
+  // lf is a log file handle and td is a dictionary with table names as keys and where clauses to filter by as values
+  // logmetatab is the tplog metadata table loaded into the stp from the tplogs directory
+    .lg.o[`subscribe;"replaying log file ",.Q.s1 lf]; -11!lf;
+  // checks if the log file contains a table that requires filtering
+    filterflags:(key td) in  raze (select from logmetatab where logname=@[lf;1])`tbls;
+    filtertab:(key td) where filterflags;
+  // filters tables replayed by the logs if required
+    if[any filterflags;
+       .lg.o[`subscribe;"filtering table(s) ", (.Q.s1 ` sv filtertab), " started at:", .Q.s1 .z.P];
+       set'[filtertab; applyfilters[;td] each filtertab];
+       .lg.o[`subscribe;"finished filtering",(.Q.s1 ` sv filtertab), " at ", .Q.s1 .z.P];
+    ];
+    };
   // set the replayupd function to be upd globally
   if[not (tabs;realsubs[`instrs])~(`;`);
     .lg.o[`subscribe;"using the .sub.replayupd function as not replaying all tables or instruments"];
     @[`.;`upd;:;.sub.replayupd[origupd;subtabs;realsubs[`instrs]]]];
-  // replays log files and applies filters if in datastriping mode
-  f:{[lf;td]
-  // lf is a log file handle and td is a dictionary with table names as keys and where clauses to filter by as values
-    .lg.o[`subscribe;"replaying log file ",.Q.s1 lf]; -11!lf;
-    if[.ds.datastripe; .[set;] each (key td),'enlist each replayfilter each (key td),' value enlist each td;  
-       .lg.o[`subscribe;"filtering tables ", .Q.s1 key td]]
-    };
-  {[d;filter;func] .[func;(d;filter);{.lg.e[`subscribe;"could not replay the log file: ", x]}]}[;filters;f] each logfilelist;
-
-
+  // if datastriping is on replays only logs from current periods and applys filtering
+    if[.ds.datastripe;
+  //get tplog metadata table from stp  
+    [logmetatab:.servers.gethandlebytype[`segmentedtickerplant;`last]`.stpm.metatable;
```

Comment:
I don't think you need the `[` here (and it's mating `]` wherever that is) - you only need that for multi-expression blocks in a `$`, not an `if`

Reply:
Yes not sure why that was like that, changed it now